src/dstack/_internal/cli/utils/rich.py

```diff
@@ -0,0 +1,119 @@
+import logging
+from datetime import datetime
+from typing import TYPE_CHECKING, Callable, Iterable, List, Optional, Union
+
+from rich.logging import RichHandler
+from rich.text import Text, TextType
+from rich.traceback import Traceback
+
+if TYPE_CHECKING:
+    from rich.console import Console, ConsoleRenderable, RenderableType
+    from rich.table import Table
+
+FormatTimeCallable = Callable[[datetime], Text]
+
+
+class LogRender:
+    def __init__(
+        self,
+        show_time: bool = True,
+        show_level: bool = True,
+        show_path: bool = True,
+        time_format: Union[str, FormatTimeCallable] = "[%x %X]",
+        omit_repeated_times: bool = True,
+        level_width: Optional[int] = 8,
+    ) -> None:
+        self.show_time = show_time
+        self.show_level = show_level
+        self.show_path = show_path
+        self.time_format = time_format
+        self.omit_repeated_times = omit_repeated_times
+        self.level_width = level_width
+        self._last_time: Optional[Text] = None
+
+    def __call__(
+        self,
+        console: "Console",
+        message_renderable: Iterable["ConsoleRenderable"],
+        log_time: Optional[datetime] = None,
+        time_format: Optional[Union[str, FormatTimeCallable]] = None,
+        level: TextType = "",
+        path: Optional[str] = None,
+        line_no: Optional[int] = None,
+        link_path: Optional[str] = None,
+    ) -> "Table":
+        from rich.containers import Renderables
+        from rich.table import Table
+
+        output = Table.grid(padding=(0, 1))
+        output.expand = True
+        if self.show_time:
+            output.add_column(style="log.time")
+        if self.show_level:
+            output.add_column(style="log.level", width=self.level_width)
+        output.add_column(ratio=1, overflow="ignore")
```

Comment:
Can't see the logs in a narrow terminal window, messages are cut. Maybe this line is the reason
![trimmed-logs](https://github.com/dstackai/dstack/assets/36324149/61a8995e-9012-4ed4-825b-9677b913924f)